nexus/tests/integration_tests/endpoints.rs

```diff
@@ -676,6 +676,21 @@ lazy_static! {
                 ),
             ],
         },
+        VerifyEndpoint {
+            url: "/policy",
+            visibility: Visibility::Protected,
+            unprivileged_access: UnprivilegedAccess::None,
+            allowed_methods: vec![
+                AllowedMethod::Get,
+                AllowedMethod::Put(
+                    serde_json::to_value(
+                        &shared::Policy::<authz::SiloRole> {
+                            role_assignments: vec![]
+                        }
+                    ).unwrap()
+                ),
+            ],
+        },
```

Comment:
This fails with the following.

```
G GET  PUT  POST DEL  TRCE G  URL
0thread 'integration_tests::unauthorized::test_unauthorized' panicked at 'called `Result::unwrap()` on an `Err` value: expected status code 404 Not Found, found 200 OK
```

It's making a request with the unprivileged user that it expects to 404 but it's 200ing instead.

https://github.com/oxidecomputer/omicron/blob/000333bab8e89987ee3a73e2780c391bd6685450/nexus/tests/integration_tests/unauthorized.rs#L444-L457

Reply:
Given the `/policy` API is known and documented if you didn't have access to that it'd more likely be a 401. I think you want to change the `visibility` in your test to `public` instead of `protected`. That might not be the original issue, but it is one thing.

Reply:
First pass at debugging using the lovely [guide](https://github.com/oxidecomputer/omicron/blob/main/docs/debugging-authz.adoc#request_succeeded). Seems like the unprivileged user does have the permission in question, which is weird?

```
~/oxide/omicron $ cat /var/folders/lk/vsx86g0545g3g2tsmvs50py40000gn/T/test_all-af18dffa1640b050-test_unauthorized.49638.0.log | rg "authorize result" | rg "/policy" | rg "ReadPolicy" | rg "60001" | jq '{ msg, req_id, authenticated, method, uri, actor, action, result }'               
{
  "msg": "authorize result",
  "req_id": "49c6298b-e273-41f4-b3ac-a79de7259074",
  "authenticated": true,
  "method": "GET",
  "uri": "/global/policy",
  "actor": "Some(Actor::SiloUser { silo_user_id: 001de000-05e4-4000-8000-000000060001, silo_id: 001de000-5110-4000-8000-000000000000, .. })",
  "action": "ReadPolicy",
  "result": "Err(Forbidden)"
}
{
  "msg": "authorize result",
  "req_id": "9b1a600f-70bc-4fd8-8103-5bde60c970bc",
  "authenticated": true,
  "method": "GET",
  "uri": "/silos/demo-silo/policy",
  "actor": "Some(Actor::SiloUser { silo_user_id: 001de000-05e4-4000-8000-000000060001, silo_id: 001de000-5110-4000-8000-000000000000, .. })",
  "action": "ReadPolicy",
  "result": "Err(ObjectNotFound { type_name: Silo, lookup_type: ByName(\"demo-silo\") })"
}
{
  "msg": "authorize result",
  "req_id": "2ba41eb1-bf4f-42d8-9ee4-a1fbe8899082",
  "authenticated": true,
  "method": "GET",
  "uri": "/policy",
  "actor": "Some(Actor::SiloUser { silo_user_id: 001de000-05e4-4000-8000-000000060001, silo_id: 001de000-5110-4000-8000-000000000000, .. })",
  "action": "ReadPolicy",
  "result": "Ok(())"
}
{
  "msg": "authorize result",
  "req_id": "2ba41eb1-bf4f-42d8-9ee4-a1fbe8899082",
  "authenticated": true,
  "method": "GET",
  "uri": "/policy",
  "actor": "Some(Actor::SiloUser { silo_user_id: 001de000-05e4-4000-8000-000000060001, silo_id: 001de000-5110-4000-8000-000000000000, .. })",
  "action": "ReadPolicy",
  "result": "Ok(())"
}
```

Reply:
This was the explanation of what I was seeing. The existing silo policy test was not asking about the user's own silo, which is why the unprivileged user didn't have access. Even the unpriv user has access to their own silo.

https://github.com/oxidecomputer/omicron/blob/45286415755d5ec7c0d3683addc9942832244eff/nexus/src/authz/omicron.polar#L148-L163